services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/AccessApprovalManagerImpl.java

```diff
@@ -185,4 +188,36 @@ public Count deleteBatch(UserInfo userInfo, IdList toDelete) {
 		result.setCount(new Long(accessApprovalDAO.deleteBatch(toDelete.getList())));
 		return result;
 	}
+
+	@Override
+	public BatchAccessApprovalResult getApprovalInfo(UserInfo userInfo, BatchAccessApprovalRequest batchRequest) {
+		ValidateArgument.required(userInfo, "userInfo");
+		ValidateArgument.required(batchRequest, "batchRequest");
+		ValidateArgument.required(batchRequest.getUserIds(), "BatchAccessApprovalRequest.userIds");
+		ValidateArgument.required(batchRequest.getAccessRequirementId(), "BatchAccessApprovalRequest.accessRequirementId");
+		if (!authorizationManager.isACTTeamMemberOrAdmin(userInfo)) {
+			throw new UnauthorizedException("Only ACT member may perform this action.");
+		}
+		BatchAccessApprovalResult batchResult = new BatchAccessApprovalResult();
+		List<AccessApprovalResult> list = new LinkedList<AccessApprovalResult>();
+		batchResult.setResults(list);
+
+		if (batchRequest.getUserIds().isEmpty()) {
+			return batchResult;
+		}
+
+		List<String> hasApprovals = accessApprovalDAO.getApprovedUsers(batchRequest.getUserIds(), batchRequest.getAccessRequirementId());
+		for (String userId : batchRequest.getUserIds()) {
+			AccessApprovalResult result = new AccessApprovalResult();
+			result.setAccessRequirementId(batchRequest.getAccessRequirementId());
+			result.setUserId(userId);
+			if (hasApprovals.contains(userId)) {
```

Comment:
Contains on List O(n).  Can the accessApprovalDAO.getApprovedUsers() return a HashSet instead (it is already using select distinct)?